src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -418,6 +415,42 @@ impl ScalarFunc {
     ) -> Result<Option<Cow<'a, MyDuration>>> {
         Ok(Some(Cow::Owned(MyDuration::zero())))
     }
+
+    #[inline]
+    pub fn add_date_and_duration<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let d0: Cow<'a, MyDuration> = try_opt!(self.children[0].eval_duration(ctx, row));
+        let d1: Cow<'a, MyDuration> = try_opt!(self.children[1].eval_duration(ctx, row));
+        let add = match d0.to_nanos().checked_add(d1.to_nanos()) {
+            Some(result) => result,
+            None => return Err(box_err!("add duration {} and duration {} overflow", d0, d1)),
```

Comment:
Should `tikv` return an error with error_code is `ERR_DATA_OUT_OF_RANGE(1690)` in this situation? @zz-jason

Reply:
I'm pretty sure we need to return error_code `ERR_DATA_OUT_OF_RANGE(1690)` here.